routes/html_routes.py

```diff
@@ -159,35 +155,20 @@ def add_movie(user_id):
             On error (POST): Redirects back to the add movie form with an error message.
             On GET: Renders the add movie form.
         """
-        user = data_manager.get_user_by_id(user_id)
-        if not user:
-            return render_error_page(404, "User not found.")
-
-        if request.method == 'POST':
-            movie_name = request.form.get('movie_name')
-
-            if not movie_name:
-                flash_message("Movie name is required.", "error")
-                return redirect(url_for('html_routes.add_movie', user_id=user_id))
-
-            try:
-                result = data_manager.add_movie(user_id, movie_name)
-
-                if 'error' in result:
-                    flash_message(result['error'], 'error')
-                else:
-                    flash_message(result['success'], 'success')
-            except KeyError as e:
-                flash_message(f"Missing data: {str(e)}", "error")
-                logger.error(f"KeyError in add_movie route: {e}")
-                return redirect(url_for('html_routes.add_movie', user_id=user_id))
-            except Exception as e:
-                logger.error(f"Error in add_movie route: {e}")
-                return render_error_page(500, "Failed to add movie.")
-
-        return render_template('add_movie.html', user_id=user_id)
+        if request.method == 'GET':
+            return render_template('add_movie.html', user_id=user_id)
+        movie_name = request.form.get('movie_name')
+        result = data_manager.add_movie(user_id, movie_name)
+        if 'error' in result:
+            flash_message(result['error'], 'error')
+        else:
+            flash_message(result['success'], 'success')
```

Comment:
We can do again more specific flash messages to get rid of the multipel string usage.
Can also be reused on multiple locations

```suggestion
            flash_error_message(text)
        else:
            flash_success_message(text)
```